### PR TITLE
circuits: integration: Add integration tests for `match` and `VALID MATCH MPC`

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR adds a test case suite for `VALID MATCH MPC` to the integration tests. These tests cover the bulk of matching engine edge cases: mismatched order, incorrect balances, incorrect amounts, invalid intermediate values, etc. 

### Todo
- Price protection

### Testing
- Integration and unit tests in `circuits` pass, though CI will not build here because `core` has not ingested these changes.

### CODE REVIEW

**Review by akirillo — APPROVED** [review]
lgtm at a high level, just have a quick q about the way these tests were set up

File: circuits/integration/zk_circuits/valid_match_mpc.rs [hunk 1]
```diff
@@ -172,17 +217,458 @@ fn test_valid_match_mpc_valid(test_args: &IntegrationTestArgs) -> Result<(), Str
...
+
+    if !prove_and_verify_match(witness, test_args.mpc_fabric.clone())? {
+        return Err("Failed to prove and verify match".to_string());
+    }
+
+    Ok(())
+}
+
+/// Test matching an order where the volume is limited by the balance of the buy side party
+fn test_valid_match_undercapitalized__quote_side(
```
[the remark this answers is not in the document]
Comment (joeykraut) [reply circuits/integration/zk_circuits/valid_match_mpc.rs:308]: This doesn't test the match logic (just the proof), we mock it out, and it doesn't settle the match either. I think an interesting integration test to add after this is all done is an end-to-end match like you mentioned. Don't think these assertions make sense in this particular test though, so I'll leave this for later

File: circuits/integration/zk_circuits/valid_match_mpc.rs [hunk 2]
```diff
@@ -96,72 +187,26 @@ fn setup_witness<N: MpcNetwork + Send, S: SharedValueSource<Scalar>>(
...
-        order2: allocated_order2,
         balance1: allocated_balance1,
+        amount1: allocated_amount1,
+        price1: allocated_price1,
+        order2: allocated_order2,
+        amount2: allocated_amount2,
         balance2: allocated_balance2,
+        price2: allocated_price2,
         match_res,
     })
```
[the remark this answers is not in the document]
Comment (joeykraut) [reply circuits/integration/zk_circuits/valid_match_mpc.rs:198]: Yeah this boils down to the programming model of the `MpcFabric` which allocates things by `party_id`. Essentially, the peers must run the same code so they run code that is:
```
party0_value = my_value.allocate(0 /* party_id */);
party1_value = my_value.allocate(1 /* party_id */);
```

The allocate logic is such that if the allocating party is the local party, the method returns its input. If the allocating party is the counterparty, the return value is fetched from the network.
